os/arch/arm/src/amebad/amebad_i2c.c

```diff
@@ -0,0 +1,1241 @@
+#endif
+	volatile uint8_t intstate;	/* Interrupt handshake (see enum amebad_intstate_e) */
+	int address;				/* Address used in this instantiation */
+
+	uint8_t msgc;				/* Message count */
+	struct i2c_msg_s *msgv;		/* Message list */
+	uint8_t *ptr;				/* Current message buffer */
+	uint32_t frequency;			/* Current I2C frequency */
+	int dcnt;					/* Current message length */
+	uint16_t flags;				/* Current message flags */
+
+	i2c_t *i2c_object;
+
+	/* I2C trace support */
+
+#ifdef CONFIG_I2C_TRACE
+	int tndx;					/* Trace array index */
+	clock_t start_time;			/* Time when the trace was started */
+
+	/* The actual trace data */
+
+	struct amebad_trace_s trace[CONFIG_I2C_NTRACE];
+#endif
+
+	uint32_t status;			/* End of transfer SR2|SR1 status */
+};
+
+/************************************************************************************
+ * Private Function Prototypes
+ ************************************************************************************/
+
+static inline uint32_t amebad_i2c_getreg(FAR struct amebad_i2c_priv_s *priv, uint16_t offset);
+static inline void amebad_i2c_putreg(FAR struct amebad_i2c_priv_s *priv, uint16_t offset, uint32_t value);
+static inline void amebad_i2c_modifyreg(FAR struct amebad_i2c_priv_s *priv, uint16_t offset, uint32_t clearbits, uint32_t setbits);
+static inline void amebad_i2c_sem_wait(FAR struct amebad_i2c_priv_s *priv);
+
+#ifdef CONFIG_AMEBAD_I2C_DYNTIMEO
+static useconds_t amebad_i2c_tousecs(int msgc, FAR struct i2c_msg_s *msgs);
+#endif							/* CONFIG_AMEBAD_I2C_DYNTIMEO */
+
+static inline int amebad_i2c_sem_waitdone(FAR struct amebad_i2c_priv_s *priv);
+static inline void amebad_i2c_sem_waitstop(FAR struct amebad_i2c_priv_s *priv);
+static inline void amebad_i2c_sem_post(FAR struct amebad_i2c_priv_s *priv);
+static inline void amebad_i2c_sem_init(FAR struct amebad_i2c_priv_s *priv);
+static inline void amebad_i2c_sem_destroy(FAR struct amebad_i2c_priv_s *priv);
+
+#ifdef CONFIG_I2C_TRACE
+static void amebad_i2c_tracereset(FAR struct amebad_i2c_priv_s *priv);
+static void amebad_i2c_tracenew(FAR struct amebad_i2c_priv_s *priv, uint32_t status);
+static void amebad_i2c_traceevent(FAR struct amebad_i2c_priv_s *priv, enum amebad_trace_e event, uint32_t parm);
+static void amebad_i2c_tracedump(FAR struct amebad_i2c_priv_s *priv);
+#endif							/* CONFIG_I2C_TRACE */
+
+static void amebad_i2c_setclock(FAR struct amebad_i2c_priv_s *priv, uint32_t frequency);
+static inline void amebad_i2c_sendstart(FAR struct amebad_i2c_priv_s *priv, uint8_t address);
+static inline void amebad_i2c_sendstop(FAR struct amebad_i2c_priv_s *priv);
+static inline uint32_t amebad_i2c_getstatus(FAR struct amebad_i2c_priv_s *priv);
+
+static int amebad_i2c_isr_process(struct amebad_i2c_priv_s *priv);
+
+#ifndef CONFIG_I2C_POLLED
+static int amebad_i2c_isr(int irq, void *context, FAR void *arg);
+#endif							/* !CONFIG_I2C_POLLED */
+
+void amebad_i2c_clock_enable(uint32_t base);
```

Comment:
Is there any reason not to add static with this function?

Reply:
Actually, we don't need this function, so I deleted it.